tests/test_invenio_records_rest.py

```diff
@@ -81,6 +84,18 @@ def create_record(data):
     return pid, record
 
 
+def delete_record(pid, record):
+    with db.session.begin_nested():
+        record.delete()
```

Comment:
@jirikuncar I am not sure doing the deletion automatically is the best way to do it. We would automate only part of the process. The pid creation is not automatic either. The records-rest API is creating both the pid and the record separately. It will also be complicated to detect the merging and mark the pid as *moved*.
Also the pidstore module does not depend on invenio-records (or only for the tests).